src/types/sidebar.js

```diff
@@ -15,5 +15,23 @@
  * @typedef {'annotation'|'note'|'orphan'} TabName
  */
 
+/**
```

Comment:
I've put this typing in the `sidebar` module because the related `downcastRef` utility is only available to the sidebar in its current location (`src/sidebar/util/typing`). We may wish to locate this elsewhere.